core/src/main/java/io/lionweb/language/Language.java

```diff
@@ -142,29 +142,30 @@ public Concept requireConceptByName(String name) {
         .orElse(null);
   }
 
-  public String getName() {
+  public @Nullable String getName() {
     return this.getPropertyValue("name", String.class);
   }
 
   @Override
-  public String getKey() {
+  public @Nullable String getKey() {
```

Comment:
This is a language's **key** which definitely shouldn't be `null`. (It's also not the name.)

Reply:
I agree that it _should not_ be null, however we decided to allow the representation of things in a broken state and so all sort of things could have a value that is not valid, I think

Reply:
In C#, we solve this by having different implementations of Node -- the default one strongly enforces constraints, and another one is very lose about it.

Reply:
Then I would say that:
* The interface `Node`, being the most generic, should the loosest one
* We could introduce another interface `ValidatedNode` that would be stricter. `ValidatedNode` should extend Node, if possible (I am not sure that we can apply restrictions on the parameters from @Nullable to @Nonnull, I would need to verify that)

Would this work? If so, I would create an issue for the introduction of `ValidatedNode` (or `ValidNode`? or `StrictNode`?)